### PR TITLE
Charcoal makes my tummy feel funny! (Return of the Hobo Coder)

### PR DESCRIPTION
**Pull request by Therandomhoboo**
# Document the changes in your pull request

I've returned to make changes that will upset people. Long story short, Charcoal is too based of a chem because it heals tox & purges all chems quicker than psych griefer gets banned. This was a long time coming, but this should stop people combat-busing this or using this as a failsafe against anyone that uses chems. Charcoal shouldn't be a straight up block against chems.

So I've added a disgust stack to the chem, it is 20 per tick currently which I think is a decent point because it still allows people to cleanse chems but not to completely get away scot-free if they have a lot of chems in them.

I also added a flavour line to notify the player too "You feel a bit queasy, you shouldn't have too much of this"

# Why is this good for the game?

It prevents mid-combat charcoaling to basically bypass any sense of chemical warfare, makes the game more interesting and punishes people that carry charcoal's just "in case"

# Testing
Its a couple of lines, chief. Couple-a-lines

# Wiki Documentation

The charcoal description needs updating with the following "causes disgust when ingested"

# Changelog

:cl:  

tweak: Charcoal now gives 20 disgust stacks, so be careful now medbay
rscadd: Charcoal now shows a text prompt "You feel a bit queasy, you shouldn't have too much of this"

/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
I hope this is a joke

**Comment by Therandomhoboo** [comment]
> I hope this is a joke

I never joke, I only upset

**Comment by wonderinghost** [comment]
> I hope this is a joke

Approved by the secret bath salts council hidden in discord

**Comment by Therandomhoboo** [comment]
> I hope this is a joke

Na for real, Charcoal is busted and it is A) Easily accessible from vendors and kit and B) Easily craftable. Medbay alone will carry a bottle of charcoal on them as standard. They can handle a debuff to basically a "get out of chem jail" card

**Comment by wonderinghost** [comment]
Chem machines can just synthesize it when fully upgraded

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -437,12 +437,16 @@
...
+	M.adjust_disgust(20)
 	M.adjustToxLoss(-2*REM, 0)
 	. = 1
 	for(var/datum/reagent/R in M.reagents.reagent_list)
 		if(R != src)
 			M.reagents.remove_reagent(R.type,1)
 	..()
+/datum/reagent/medicine/charcoal/reaction_mob(mob/living/M, methods=TOUCH)
+	if(methods & INGEST)
+		to_chat(M, span_notice("You feel a bit queasy, you shouldn't have too much of this"))
```
Comment (iamthesilly) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:449]: You need `return ..()` after this line to not override the normal `reaction_mob()` function

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Comment by LazennG** [comment]
griefer attempts code sabotage

**Comment by iamthesilly** [comment]
I heavily disagree with this because not only does it invalidate one of charcoal's main intended uses as a counter to chemical weapons, but also turns it into an incredibly powerful offensive chem that stunlocks you in seconds by giving you more disgust every 2 seconds than you do from crushing a bluespace crystal.

It's also the only commonly available chem that can heal toxin damage and would make it a lot harder for medics to do their job by making their patients want to reject it.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (Therandomhoboo) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:449]: whoops thanks

**Comment by Therandomhoboo** [comment]
> I heavily disagree with this because not only does it invalidate one of charcoal's main intended uses as a counter to chemical weapons, but also turns it into an incredibly powerful offensive chem that stunlocks you in seconds by giving you more disgust every 2 seconds than you do from crushing a bluespace crystal.
> 
> It's also the only commonly available chem that can heal toxin damage and would make it a lot harder for medics to do their job by making their patients want to reject it.

Like I said prior, happy to tweak the disgust stacks. However, what is frustrating is that it is freely available everywhere and it pretty much squashes anyone that want to use poisons or bad chems, this should encourage medbay players to make the better toxin heal alternative like anti-toxin which should be the point of slow escalation.

I just think this is a massive crutch for medbay users and anyone that has time in the game, it's basically a free tox+chem purge at the cost of nothing.

**Comment by cowbot92** [comment]
I would make it do probably literally anything else than disgust given how prevalent it is and how *powerful* disgust is- maybe consider OD thresholds

**Comment by iamthesilly** [comment]
> maybe consider OD thresholds

OD thresholds would still make it very easy to weaponize, and it only takes around 5 units to purge a syringe worth of chems so the threshold would need to be incredibly low

**Comment by Therandomhoboo** [comment]
> I would make it do probably literally anything else than disgust given how prevalent it is and how _powerful_ disgust is- maybe consider OD thresholds

I'm happy to turn this into OD really,

> OD thresholds would still make it very easy to weaponize, and it only takes around 5 units to purge a syringe worth of chems so the threshold would need to be incredibly low

I could swap out disgust for confusion & blurriness or just confusion so its less weaponised and then it could make it that if you're dumping it midcombat then it can end up bad.

**Comment by iamthesilly** [comment]
> I just think this is a massive crutch for medbay users and anyone that has time in the game, it's basically a free tox+chem purge at the cost of nothing.

Medbay uses it so much because it's the only roundstart option for healing toxin damage that isn't the cryo tubes.

**Comment by Therandomhoboo** [comment]
> > I just think this is a massive crutch for medbay users and anyone that has time in the game, it's basically a free tox+chem purge at the cost of nothing.
> 
> Medbay uses it so much because it's the only roundstart option for healing toxin damage that isn't the cryo tubes.

I get that, but I don't think I've seen any other non-charcoal toxin heals specifically (not including all-around heals) made just cause charcoal is abundant and super easy to make.

**Comment by ynot01** [comment]
This will make charcoal an offensive stun-chem, consider other routes or workarounds that make this not true and you will get my vote (but not guaranteed others')

**Comment by wonderinghost** [comment]
> > maybe consider OD thresholds
> 
> OD thresholds would still make it very easy to weaponize, and it only takes around 5 units to purge a syringe worth of chems so the threshold would need to be incredibly low

could change the recipe of charcoal. having to refine it to be active charcoal by heating it up, then have a base charcoal which has impurities

**Comment by Therandomhoboo** [comment]
> This will make charcoal an offensive stun-chem, consider other routes or workarounds that make this not true and you will get my vote (but not guaranteed others')

Would of been happy to give lower disgust stacks, replace it with confusion &/or blurry or give temp slowdown. We shouldn't be encouraging players to basically sit on their ass and anytime they get hit by a chem they don't like just down some bottle of charcoal and be completely fine.

No one makes anti-toxin, Calomel (because it does toxin damage), Haloperidol or any others that are meant to be the "better" versions of them. Because Charcoal does the job, heals toxin and is easily craftable. Even when I played as chemist and would go to make them, CMO or others would just say make Charcoal

During any of my playtime there, highpop, medpop or lowpop I don't see any other purge chemicals nor any other toxin heals. So yeah, it needs to be sorted because what's the point of having poisons or offensive chems if all you have to do is eat ash and water?

Also, I can always tweak the flush tick down so we can actual use other purges. (from 2 down to 0.75 or 1 personally)

**Pull request closed without merging**